doc/nrf/libraries/others/event_manager.rst

```diff
@@ -302,27 +302,58 @@ The Event Manager provides additional features that could be helpful when debugg
 
 .. _event_manager_profiling_tracing_hooks:
 
+Initialization hook
+===================
+
+.. em_initialization_hook_start
+
+Event manager provides initialization hook for any modules that relays on the fact that Event manager is initialized
+but it should be initialized before the first event is processed.
+The hook function should be declared in a form `int hook(void)`.
+If the hook function returns non zero value, initialization process would be interrupted and the given error would be returned.
+
+To register such a hook use prepared macro:
+
+* :c:macro:`EVENT_MANAGER_HOOK_POSTINIT_REGISTER`
+
+For details, refer to `API documentation`_.
+
+.. em_initialization_hook_end
+
 Tracing hooks
 =============
 
 .. em_tracing_hooks_start
 
-Event Manager provides tracing hooks that you can use at run time to get information about Event Manager initialization, event submission, and event execution.
-The hooks are provided as weak functions.
-You can override them for interacting with a custom profiler or for other purposes.
+Event manager uses flexible mechanism to implement hooks when the event is signed, before it is processed,
+and after its processing.
+Oryginally designed to implement event tracing may be used to any different purposes.
+The registered hook function should be in a form `void hook(const struct event_header *eh)`.
 
-The following weak functions are provided by Event Manager as hooks:
+The following macros was implemented to register event tracing hooks:
 
-* :c:func:`event_manager_trace_event_execution`
-* :c:func:`event_manager_trace_event_submission`
-* :c:func:`event_manager_trace_event_init`
-* :c:func:`event_manager_alloc`
-* :c:func:`event_manager_free`
+* :c:macro:`EVENT_MANAGER_HOOK_POSTINIT_REGISTER`
+* :c:macro:`EVENT_HOOK_ON_SUBMIT_REGISTER_FIRST`, :c:macro:`EVENT_HOOK_ON_SUBMIT_REGISTER`, :c:macro:`EVENT_HOOK_ON_SUBMIT_REGISTER_LAST`
+* :c:macro:`EVENT_HOOK_PREPROCESS_REGISTER_FIRST`, :c:macro:`EVENT_HOOK_PREPROCESS_REGISTER`, :c:macro:`EVENT_HOOK_PREPROCESS_REGISTER_LAST`
+* :c:macro:`EVENT_HOOK_POSTPROCESS_REGISTER_FIRST`, :c:macro:`EVENT_HOOK_POSTPROCESS_REGISTER`, :c:macro:`EVENT_HOOK_POSTPROCESS_REGISTER_LAST`
 
 For details, refer to `API documentation`_.
 
 .. em_tracing_hooks_end
 
+Memory management hooks
+=======================
+
+Event Manager implements default memory management functions using weak implementation.
```

Comment:
Write less implementation details in documentation because if anything changes, we can forget to update it here. If the user is interested in implementation details, he can go to implementation. In the end - it is source code, not library.